subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r85/TestFailureProgressEventCrossVersionTest.groovy

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.tooling.r85
+
+import org.gradle.integtests.tooling.fixture.TargetGradleVersion
+import org.gradle.integtests.tooling.fixture.TestFailureSpecification
+import org.gradle.integtests.tooling.fixture.ToolingApiVersion
+import org.gradle.tooling.BuildException
+import org.gradle.tooling.TestAssertionFailure
+
+@ToolingApiVersion(">=8.5")
+@TargetGradleVersion(">=8.5")
+class TestFailureProgressEventCrossVersionTest extends TestFailureSpecification {
```

Comment:
Could you create a test case with comparison failure for Junit 4 and Junit 5?
```
    def "Test failure contains mapped test comparison causes using JUnit 5"() {
        given:
        setupJUnit5()
        file('src/test/java/org/gradle/JUnitTest.java') << '''
            package org.gradle;
            import org.junit.jupiter.api.Test;
            public class JUnitTest {
                @Test
                void test() {
                    throw new AssertionError(
                        "This exception wraps an assertion error",
                        new AssertionFailedError("This is a wrapped assertion error", "Expected text", "Actual text")
                    );
                }
            }
        '''
        def collector = new TestFailureEventCollector()

        when:
        runTestTaskWithFailureCollection(collector)

        then:
        thrown(BuildException)
        collector.failures.size() == 1
        collector.failures[0] instanceof TestFailure

        TestFailure failure = collector.failures[0]
        failure.message == "This exception wraps an assertion error"

        failure.causes.size() == 1
        failure.causes[0] instanceof TestAssertionFailure
        
        TestAssertionFailure assertionFailure = failure.causes[0]
        assertionFailure.message == "This is a wrapped assertion error"
        assertionFailure.expected == "Expected text"
        assertionFailure.actual == "Actual text"
    }
```

Reply:
Yeah, can do!
Also another case which I missed: is the functionality correct, when the assertion exception is wrapped in a runtime exception?

Reply:
Added two new test cases